instrument.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tally
+
+import (
+	"fmt"
+)
+
+// NewInstrumentedCall returns an InstrumentedCall with the given name
+func NewInstrumentedCall(scope Scope, name string) InstrumentedCall {
+	return &instrumentedCall{
+		error:   scope.Tagged(map[string]string{"result_type": "error"}).Counter(name),
```

Comment:
nit: use constants instead of inline strings

i.e. add the following to the top of the file 
```
const (
  resultTypeString = "result_type" 
) 
```

also, use a enum type for error/success, something like 
```
type resultType string 

const (
  resultTypeSuccess resultType = iota
  resultTypeError
)
```

Reply:
Won't be able to use an enum type as it tag values need to be a string, but you can use:
```go
type resultType string

const (
  successResultType resultType = "success"
  errorResultType resultType = "error"
)
```

And for the actual tag name probably best to name it as such (indicating its a tag name):
```go
const resultTypeTagName = "result_type"
```

Reply:
Sorry, I am failing to see the purpose of the enum type. Having `0` and `1` isn't very human readable for m3 tags?

Reply:
sorry, meant to use the string version @robskillington suggested, `iota` was muscle memory